src/signax/signatures.py

```diff
@@ -49,28 +111,32 @@ def _body(carry, path_inc):
     carry, stacked = jax.lax.scan(f=_body, init=exp_term, xs=path_increments[1:])
 
     if stream:
-        return [
+        res = [
             jnp.concatenate([first[None, ...], rest], axis=0)
             for first, rest in zip(exp_term, stacked)
         ]
-
-    return carry
+    else:
+        res = carry
+    if flatten:
```

Comment:
Since the shape of `res` when `stream=True` (shape = `[(path_len-1, dim), (path_len-1, dim, dim]`) is different to when `stream=False` (shape = `[(dim,), (dim, dim)]`), we may use `jax.vmap(utils.flatten)` when `stream=True`.  

We vmap the first dimesion `path_len - 1`

Reply:
covered this by using `ravel_pytree` and getting rid of `flatten`:

```python3
import jax.numpy as jnp
from jax import flatten_util

arr = [jnp.ones((2, 3)), jnp.ones((2, 3, 4))]
tree = flatten_util.ravel_pytree(arr)[0]
print(tree)
# > [1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]
```